Add support for empty path prefixes

If I set path prefixes dynamically using `PrefixPathGatewayFilterFactory`, and one of them happens to be an empty string, I get an exception
```
Caused by: java.lang.IllegalArgumentException: 'uriTemplate' must not be null
```
It's because its `apply()` method calls the `UriTemplate` constructor
```java
	@Override
	public GatewayFilter apply(Config config) {
		return new GatewayFilter() {
			final UriTemplate uriTemplate = new UriTemplate(config.prefix);
```
which in turn checks the prefix for emptiness (not just nullness, as the exception message suggests)
```java
	public UriTemplate(String uriTemplate) {
		Assert.hasText(uriTemplate, "'uriTemplate' must not be null");
```
It's super-easy to fix: simply replace
```java
Assert.hasText(uriTemplate, "'uriTemplate' must not be null");
```
with
```java
Assert.notNull(uriTemplate, "'uriTemplate' must not be null");
```

--- a/spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/filter/factory/PrefixPathGatewayFilterFactory.java
+++ b/spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/filter/factory/PrefixPathGatewayFilterFactory.java
@@ -20,6 +20,7 @@ import java.net.URI;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
@@ -62,7 +63,8 @@ public class PrefixPathGatewayFilterFactory
 	@Override
 	public GatewayFilter apply(Config config) {
 		return new GatewayFilter() {
-			final UriTemplate uriTemplate = new UriTemplate(config.prefix);
+			final UriTemplate uriTemplate = new UriTemplate(
+					Objects.requireNonNull(config.prefix, "prefix must not be null"));
 
 			@Override
 			public Mono<Void> filter(ServerWebExchange exchange, GatewayFilterChain chain) {
@@ -91,7 +93,7 @@ public class PrefixPathGatewayFilterFactory
 
 			@Override
 			public String toString() {
-				return filterToStringCreator(PrefixPathGatewayFilterFactory.this).append("prefix", config.getPrefix())
+				return filterToStringCreator(PrefixPathGatewayFilterFactory.this).append(PREFIX_KEY, config.getPrefix())
 					.toString();
 			}
 		};

--- a/spring-cloud-gateway-server/src/test/java/org/springframework/cloud/gateway/filter/factory/PrefixPathGatewayFilterFactoryTest.java
+++ b/spring-cloud-gateway-server/src/test/java/org/springframework/cloud/gateway/filter/factory/PrefixPathGatewayFilterFactoryTest.java
@@ -22,6 +22,7 @@ import java.util.LinkedHashSet;
 
 import org.junit.jupiter.api.Test;
 import org.mockito.ArgumentCaptor;
+import org.springframework.web.util.UriTemplate;
 import reactor.core.publisher.Mono;
 
 import org.springframework.cloud.gateway.filter.GatewayFilter;
@@ -31,8 +32,11 @@ import org.springframework.cloud.gateway.support.ServerWebExchangeUtils;
 import org.springframework.mock.http.server.reactive.MockServerHttpRequest;
 import org.springframework.mock.web.server.MockServerWebExchange;
 import org.springframework.web.server.ServerWebExchange;
+import wiremock.org.apache.commons.lang3.StringUtils;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatNoException;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 import static org.springframework.cloud.gateway.support.ServerWebExchangeUtils.GATEWAY_ORIGINAL_REQUEST_URL_ATTR;
@@ -96,6 +100,19 @@ public class PrefixPathGatewayFilterFactoryTest {
 		config.setPrefix("myprefix");
 		GatewayFilter filter = new PrefixPathGatewayFilterFactory().apply(config);
 		assertThat(filter.toString()).contains("myprefix");
+	}
+
+	@Test
+	public void testNullPrefixThrowsException() {
+		PrefixPathGatewayFilterFactory factory = new PrefixPathGatewayFilterFactory();
+		assertThatThrownBy(() -> factory.apply(c -> c.setPrefix(null)))
+				.isInstanceOf(NullPointerException.class)
+				.hasMessage("prefix must not be null");
+	}
+
+	@Test
+	public void emptyPathDoesNotThrowException() {
+		assertThatNoException().isThrownBy(() -> new UriTemplate(StringUtils.EMPTY));
 	}
 
 }

